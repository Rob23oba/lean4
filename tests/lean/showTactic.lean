example : (∀ x, x = 0) ∧ (∀ x, x = 1) ∧ (∀ x, x = 2) ∧ (∀ x, x = 3) := by
  and_intros
  all_goals intro x
  show _ = 0 + 0
  trace_state
  show _ = 3
  trace_state
  show _ = 2
  show _ = 1
  show _ = 0
  trace_state
  show _ = 4

example : (∀ x, x = 0) ∧ (∀ y, y = 1) ∧ (∀ z, z = 2) ∧ (∀ a, a = 3) := by
  and_intros
  all_goals unhygienic intro
  show _ + 0 = 2
  trace_state
  show a = _
  trace_state
  show _ = 0
  trace_state
  show b = 3
