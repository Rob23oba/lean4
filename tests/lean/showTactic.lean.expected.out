showTactic.lean:12:2-12:12: error: tactic 'show' failed, nested error:
'change' tactic failed, pattern
  ?m = 4
is not definitionally equal to target
  x = 3
case refine_1
x : Nat
⊢ x = 0 + 0

case refine_2.refine_1
x : Nat
⊢ x = 1

case refine_2.refine_2.refine_1
x : Nat
⊢ x = 2

case refine_2.refine_2.refine_2
x : Nat
⊢ x = 3
case refine_2.refine_2.refine_2
x : Nat
⊢ x = 3

case refine_1
x : Nat
⊢ x = 0 + 0

case refine_2.refine_1
x : Nat
⊢ x = 1

case refine_2.refine_2.refine_1
x : Nat
⊢ x = 2
case refine_1
x : Nat
⊢ x = 0

case refine_2.refine_1
x : Nat
⊢ x = 1

case refine_2.refine_2.refine_1
x : Nat
⊢ x = 2

case refine_2.refine_2.refine_2
x : Nat
⊢ x = 3
showTactic.lean:23:7-23:8: error: unknown identifier 'b'
showTactic.lean:14:71-23:12: error: unsolved goals
case refine_1
x : Nat
⊢ sorry = 3

case refine_2.refine_2.refine_2
a : Nat
⊢ a = 3

case refine_2.refine_2.refine_1
z : Nat
⊢ z + 0 = 2

case refine_2.refine_1
y : Nat
⊢ y = 1
case refine_2.refine_2.refine_1
z : Nat
⊢ z + 0 = 2

case refine_1
x : Nat
⊢ x = 0

case refine_2.refine_1
y : Nat
⊢ y = 1

case refine_2.refine_2.refine_2
a : Nat
⊢ a = 3
case refine_2.refine_2.refine_2
a : Nat
⊢ a = 3

case refine_2.refine_2.refine_1
z : Nat
⊢ z + 0 = 2

case refine_1
x : Nat
⊢ x = 0

case refine_2.refine_1
y : Nat
⊢ y = 1
case refine_1
x : Nat
⊢ x = 0

case refine_2.refine_2.refine_2
a : Nat
⊢ a = 3

case refine_2.refine_2.refine_1
z : Nat
⊢ z + 0 = 2

case refine_2.refine_1
y : Nat
⊢ y = 1
